### DISCUSSION TITLE
ATTiny Series 1 TWI Repeated Start

### DISCUSSION THREAD

**Discussion by ghost** [SpenceKonde/megaTinyCore#1083] [Q&A] [answered]
I don't see in the TWI (Wire) code where the Repeated Start command is used.  The datasheet for the ATTIny1614/1616/1617 shows one of the commands for MCTRLB is REPSTART, but the datasheet doesn't indicate if this command is issued BEFORE the MADDR slave address is set or AFTER.   E.g., here are two possibilities:

// Configure Repeated Start:
TWI0.MCTRLB = TWI_MCMD_REPSTART_gc;
// Send Slave address.
TWI0.MADDR = slaveAddress | direction;

OR

// Send Slave address.
TWI0.MADDR = slaveAddress | direction;
// Configure Repeated Start:
TWI0.MCTRLB = TWI_MCMD_REPSTART_gc;

Has anyone implemented this command?
Thanks :)

**Comment by MX682X** [comment, marked as answer]
When implementing the Wire Library I noticed that it doesn't seem to matter if START or REPSTART is issued, the Hardware automatically does a REPSTART if there was no STOP condition.
And as the START command is issued anytime the MADDR register is written, I don't see any practical use of the REPSTART command, except on continous writing to the same slave, which again can be implemented by just writing the same Address to the MADDR Register.
It is entioned that the CMD in MCTRLB is a strobe, thus executed immediately, like the STOP Command.

**Reply by ghost** [reply]
Thank you.
